core/types/block.go

```diff
@@ -66,6 +66,14 @@ func (n *BlockNonce) UnmarshalText(input []byte) error {
 	return hexutil.UnmarshalFixedText("BlockNonce", input, n[:])
 }
 
+// StateData represents state received from Ethereum Blockchain
+type StateData struct {
```

Comment:
Create new `core/types/state-data.go` file for struct. It's better we don't touch block.go